src/runge-kutta.jl

```diff
@@ -0,0 +1,315 @@
+# This file contains the implementation of explicit Runkge-Kutta
+# solver from (Hairer & Wanner 1992 p.134, p.165-169).
+
+include("tableaus.jl")
+
+# intermediate level interface
+
+"""
+
+A general Runge-Kutta stepper (it cen represent either, a fixed step
+or an adaptive step algorithm).
+
+"""
+immutable RKStepper{Kind,tab_name,T<:Number} <: AbstractStepper
+    tableau::Tableau
+    function RKStepper()
+        tab = eval(tab_name)
```

Comment:
I already removed the `S` and `Name` (although I haven't commit the changes yet).  This way you can specify all the type parameters (only one remaining) in `Tableau` in `RKStepper`.  I don't think that the `Name` in `Tableau` will ever be necessary now that we have the stepper as a type.  The `RKStepper` looks like this

```
immutable RKStepper{Kind,Name,T} <: AbstractStepper{T}
    tableau::Tableau{T}
    function RKStepper()
        tab = convert(T,eval(Name))
        if Kind == :fixed && isadaptive(tab)
            error("Cannot construct a fixed step method from an adaptive step tableau")
        elseif Kind == :adaptive && !isadaptive(tab)
            error("Cannot construct an adaptive step method from an fixed step tableau")
        end
        new(tab)
    end
end
```

and you can find the `Tableau` is here https://gist.github.com/pwl/50c705538d6c86087180f99fac1bfe68 (sorry, no `Dict` yet, I can add it later).

I can still revert the changes although to me the code looks less cluttered now.  I'm not commiting anything yet because I am still looking for a fix to the type instability.